crates/vm2/src/lib.rs

```diff
@@ -65,6 +66,43 @@ pub trait StorageInterface {
     fn is_free_storage_slot(&self, contract: &H160, key: &U256) -> bool;
 }
 
+impl StorageInterface for Infallible {
+    fn read_storage(&mut self, _contract: H160, _key: U256) -> Option<U256> {
+        unreachable!("`Infallible` cannot be constructed")
```

Comment:
```suggestion
        match *self {}
```

Why panic when you can invoke [Ex falso quodlibet](https://en.wikipedia.org/wiki/Principle_of_explosion) instead?